src/model.h

```diff
@@ -0,0 +1,174 @@
+#ifndef MODEL_H
+#define MODEL_H
+
+#include "logging.h"
+#include "types.h"
+
+#include <opencv2/calib3d.hpp>
+#include <opencv2/core.hpp>
+#include <opencv2/highgui.hpp>
+#include <opencv2/imgproc.hpp>
+
+#include <map>
+
+namespace var {
+
+class model {
+public:
+    model(const std::string& path) {
+        _pitch = cv::imread(path);
+        _pitchSave = _pitch.clone();
+
+        cv::cvtColor(_pitch, _pitch, cv::COLOR_BGR2RGBA);
+    }
+
+    std::map<int, cv::Point2f>& imagePoints() { return _imagePoints; }
+    std::map<int, cv::Point2f>& pitchPoints() { return _pitchPoints; }
+    std::string& imagePath() { return _imagePath; }
+    cv::Mat& image() { return _image; }
+    cv::Mat& pitch() { return _pitch; }
+    vec2& pitchSize() { return _pitchSize; }
+    vec2& attackerPos() { return _attackerPos; }
+    vec2& defenderPos() { return _defenderPos; }
+
+    void load() {
+        _image = cv::imread(_imagePath);
+        _imageSave = _image.clone();
+
+        cv::cvtColor(_image, _image, cv::COLOR_BGR2RGBA);
+    }
+
+    void save(const std::string& path) {
+        try {
+            auto [image, _] = generate();
+
+            cv::cvtColor(image, image, cv::COLOR_RGBA2BGR);
+            cv::imwrite(path, image);
+        } catch (const std::exception& ex) {
+            logging::error("{}", ex.what());
+        }
+    }
+
+    std::pair<cv::Mat, cv::Mat> generate() {
```

Comment:
Consider splitting into two (private) functions: transformationg and line drawing.